src/components/Settings.vue

```diff
@@ -57,7 +71,35 @@ export default {
 		displayUserHandle() {
 			return this.$store.state.handle;
 		},
-		changePassword() {}
+		changePassword() {
+			let self = this;
+			var user = firebase.auth().currentUser;
+			var credential = firebase.auth.EmailAuthProvider.credential(
+				this.$store.state.userId,
+				this.oldPassword
+			);
+
+			user.reauthenticateWithCredential(credential)
+				.then(function() {
+					// User re-authenticated.
+					user.updatePassword(self.newPassword)
+						.then(function() {
+							console.log("Password update successful!");
+						})
+						.catch(function(error) {
+							console.log(
+								"An error occcurred while changing the password:",
+								error
+							);
+						});
```

Comment:
I would put a TODO to add a toast letting the user know the change was successful